### PR TITLE
Add connection attributes to sqlalchemy connect span

### PR DESCRIPTION
**Pull request by bschoenmaeckers**
# Description

The connect span did not have the same attributes as other sqlalchemy spans. This MR adds the same connection attributes as the other spans extracted out of the connection url.

Related to #1134

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

I have updated the tests accordingly.

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Review by srikanthccv — APPROVED** [review]

**Comment by shalevr** [comment]
Please fix the changelog conflict

**Comment by bschoenmaeckers** [comment]
Conflict fixed

**Review by shalevr — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -29,6 +29,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#1592](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1592))
 - `opentelemetry-instrumentation-django` Allow explicit `excluded_urls` configuration through `instrument()`
   ([#1618](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1618))
+- Add connection attributes to sqlalchemy connect span
+  ([#1608](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1608))
```
Comment (srikanthccv) [inline CHANGELOG.md:33]: Move this to `## Unreleased ###Added section`

**Comment by bschoenmaeckers** [comment]
Fixed formatting. Please run checks again.

**Pull request merged**